lib/artirix_data_models/cached_action_adaptor.rb

```diff
@@ -44,10 +44,10 @@ def enabled?
   private
 
   def get_cached_result
-    return nil unless enabled?
+    raise "Unexpected NilClass returned from cache - case 2" unless enabled?
 
     c = cache_read
-    return nil unless c.present?
+    raise "Unexpected NilClass returned from cache - case 3" unless c.present?
```

Comment:
all in all, if you think that the cache is returning bad value (`nil` that shouldn't be a `nil`), the easiest way of checking it is by looking at the logs. We do log in `debug` each time we check that a cache key EXISTS, every time that we READ and every time that we WRITE.

https://github.com/adrdilauro/artirix_data_models/blob/a677b963d4df0d3c8134babe656b24f3c6705b12/lib/artirix_data_models/cached_action_adaptor.rb#L96
https://github.com/adrdilauro/artirix_data_models/blob/a677b963d4df0d3c8134babe656b24f3c6705b12/lib/artirix_data_models/cached_action_adaptor.rb#L104
https://github.com/adrdilauro/artirix_data_models/blob/a677b963d4df0d3c8134babe656b24f3c6705b12/lib/artirix_data_models/cached_action_adaptor.rb#L111